rearrange chomp modules for maintainability

rhaschke opened the pull request:
Cleaning up chomp sources that are scattered around the repository, creating a circular dependency between chomp, moveit_core, and moveit_experimental.
This fixes #1136:
- move `collision_distance_field` from `moveit_experimental` back to `moveit_core`
- move `chomp_optimizer_adapter.cpp` into own package `moveit_chomp_optimizer_adapter`
- rename `default_planner_request_adapters/CHOMPOptimizerAdapter` -> `chomp/OptimizerAdapter`

v4hn commented on moveit_planners/chomp/chomp_optimizer_adapter/chomp_optimizer_adapter_plugin_description.xml:
```diff
@@ -0,0 +1,9 @@
+<library path="libmoveit_chomp_optimizer_adapter">
+
+  <class name="chomp/OptimizerAdapter" type="chomp::OptimizerAdapter"
```
:+1: for the name change. It's simply not a "default".

[The tutorial](https://ros-planning.github.io/moveit_tutorials/doc/chomp_planner/chomp_planner_tutorial.html) has to be changed to the new name though.

rhaschke replied:
This is on my (long) list...

rhaschke commented on moveit_planners/chomp/chomp_optimizer_adapter/CMakeLists.txt:
```diff
@@ -0,0 +1,39 @@
+#  LIBRARIES chomp_optimizer_adapter
+#  CATKIN_DEPENDS other_catkin_pkg
+#  DEPENDS system_lib
+)
```
Done.

rhaschke commented on moveit_core/collision_distance_field/CMakeLists.txt:
```diff
@@ -9,23 +9,22 @@ add_library(${MOVEIT_LIB_NAME}
 
-add_library(collision_detector_hybrid_plugin src/collision_detector_hybrid_plugin_loader.cpp)
-set_target_properties(collision_detector_hybrid_plugin PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
-target_link_libraries(collision_detector_hybrid_plugin ${catkin_LIBRARIES} ${MOVEIT_LIB_NAME})
```
Header and source are removed too.

rhaschke commented:
@v4hn I fixed the remaining issues and addressed your comments. Do you want to have a final look?
I would like to continue with the release as announced.

v4hn commented on moveit_experimental/CMakeLists.txt:
```diff
@@ -56,26 +45,17 @@ catkin_package(
   DEPENDS
     EIGEN3
     Boost
+    OCTOMAP
```
> moveit_experimental doesn't build anything anymore...

True.
It feels stupid to keep all the dependencies for a module that does not build anything.
Shouldn't we at least comment out all the unnecessary bits?
They cost build time and are simply baggage at this point.

rhaschke replied:
I will be fine with this ;-)

v4hn commented:
I just stumbled over https://travis-ci.org/ros-planning/moveit_visual_tools/jobs/465994691#L479
on the build farm.
Was this only a temporary problem or are the shadow-fixed debs broken for now?

@rhaschke

rhaschke commented:
I was so brave to update the binary sources on my machine. There I get a more verbose error msg:
`E: /tmp/apt-dpkg-install-i6AXUC/02-ros-melodic-moveit-core_0.10.6-0bionic.20181210.112240_amd64.deb: trying to overwrite '/opt/ros/melodic/include/moveit/collision_distance_field/collision_common_distance_field.h', which is also in package ros-melodic-moveit-experimental 0.10.5-0bionic.20181117.194840`

So, we need to "uninstall" the obsolete binary package `ros-melodic-moveit-experimental`.
@clalancette Is there an option to do so automatically? Manually doing it, resolved the problem locally.

For Travis, I guess we need to wait for dockerhub to rebuild our docker image:
https://hub.docker.com/r/moveit/moveit/builds

However, as `ci-shadow-fixed` builds on top of `ci`, this will probably fail too.

rhaschke commented:
@v4hn I guess, we should back-port this to Kinetic as well. What do you think?
